app/services/user-theme.js

```diff
@@ -0,0 +1,92 @@
+import Ember from 'ember';
+import C from 'ui/utils/constants';
+
+const AUTO_UPDATE_TIMER = 1800000; // 30min
+
+export default Ember.Service.extend({
+  prefs : Ember.inject.service(),
+
+  defaultUiTheme : null,
+
+  themeObserver: function() {
+    if (this.get('defaultUiTheme') === 'ui-auto') {
+      this.setAutoUpdate();
+    }
+  }.observes('defaultUiTheme'),
+
+  setupTheme: function() {
+    var userTheme    = this.get(`prefs.${C.PREFS.DEFAULT_UI_THEME}`);
+    var defaultTheme = this.get('defaultUiTheme');
+
+    // does the uitheme exist in prefs
+    if (userTheme) {
+
+      // dooes the user pref'd theme match the default theme from local storage
+      if (userTheme !== defaultTheme) {
+
+        if (defaultTheme !== 'ui-auto') {
+          this.set(`prefs.${C.PREFS.DEFAULT_UI_THEME}`, userTheme);
+        } else {
+          this.setAutoUpdate();
+        }
+
+        window.localStorage.setItem('defaultUiTheme', userTheme);
+
+        this.set('defaultUiTheme', userTheme);
+      }
+    } else { // no user pref'd theme
+      this.set(`prefs.${C.PREFS.DEFAULT_UI_THEME}`, defaultTheme);
+    }
+  },
+
+  setTheme: function(newTheme) {
+
+    this.set(`prefs.${C.PREFS.DEFAULT_UI_THEME}`, newTheme);
+
+    this.writeStyleNode(newTheme);
+
+    this.set('defaultUiTheme', newTheme);
+
+  },
+
+  getTheme: function() {
+    return this.get(`prefs.${C.PREFS.DEFAULT_UI_THEME}`);
+  },
+
+  setAutoUpdate: function() {
+
+    var hour     = new Date().getHours();
+    var newTheme = 'ui-light';
+
+    if ((hour >= 7 && hour <= 14)) {
+
+      this.writeStyleNode(newTheme);
+    } else {
+
+      newTheme = 'ui-dark';
+
+      this.writeStyleNode(newTheme);
+    }
+
+    this.set('updateTimer', Ember.run.later(() => {
+      return this.setAutoUpdate();
+    }, AUTO_UPDATE_TIMER));
```

Comment:
For extra flair, if the hour is near the cutoffs the interval should be the right number of minutes so it fires actually on the hour (and no minutes) instead of at potentially 6:59am->7:29am.